npm/private/npm_translate_lock.bzl

```diff
@@ -550,24 +556,17 @@ def _impl(rctx):
         )
 
     # in Bazel 5.3.0, the lock file workspace_name will now be empty if it is the local workspace;
-    # we use rctx.attr.link_workspace instead to handle the explicit link workspace name case (e2e/rules_foo for example);
+    # we use rctx.attr.link_workspace instead to handle the explicit link workspace name case (e2e/rules_foo for example)
     # check that there isn't conflicting link workspaces if both are set
-    if link_workspace and rctx.attr.link_workspace and link_workspace != rctx.attr.link_workspace:
-        msg = "lock file workspace_name '{}' and link_workspace '{}' are both set but do not match".format(
-            link_workspace,
-            rctx.attr.link_workspace,
-        )
-        fail(msg)
-
-    if not link_workspace:
+    if rctx.attr.link_workspace:
```

Comment:
Let me know if you end up changing this at all...

Reply:
Looked at this again. The change here is correct. I was just missing setting link_workspace in the downstream rule sets.